File: tutorials/golang/05-hello-concurrency/main.go
package main

import (
	"fmt"
	"net/http"
	"time"
)

func main() {
	urls := []string{
		"https://httpbin.org/delay/1",
		"https://httpbin.org/delay/1",
		"https://httpbin.org/delay/1",
		"https://httpbin.org/status/404",
		"https://httpbin.org/get",
	}

	client := &http.Client{Timeout: 10 * time.Second}

	fmt.Printf("Fetching %d URLs concurrently...\n", len(urls))
	start := time.Now()
	results := FetchAll(urls, client)
	total := time.Since(start)

	PrintResults(results)
	fmt.Printf("\nTotal time: %s (sequential would be ~%ds)\n", total, len(urls))
}

File: tutorials/rust/05-hello-concurrency/src/main.rs
use hello_concurrency::fetch_all;

#[tokio::main]
async fn main() {
    let urls = vec![
        "https://httpbin.org/get".to_string(),
        "https://httpbin.org/ip".to_string(),
        "https://httpbin.org/user-agent".to_string(),
    ];

    println!("Fetching {} URLs concurrently...", urls.len());
    let results = fetch_all(urls).await;

    for result in results {
        match result {
            Ok(r) => println!("  {} -> {} ({} bytes)", r.url, r.status, r.size),
            Err(e) => println!("  Error: {}", e),
        }
    }
}
